modules/repository/create.go

```diff
@@ -39,6 +39,8 @@ func CreateRepository(doer, u *models.User, opts models.CreateRepoOptions) (_ *m
 		IsPrivate:                       opts.IsPrivate,
 		IsFsckEnabled:                   !opts.IsMirror,
 		CloseIssuesViaCommitInAnyBranch: setting.Repository.DefaultCloseIssuesViaCommitsInAnyBranch,
+		IsHookTaskPurgeEnabled:          setting.Repository.DefaultIsHookTaskPurgeEnabled,
```

Comment:
do we real have to set it on repo creation?
instead of use global default until user change it?

Reply:
I guess I was implementing the way "CloseIssuesViaCommitInAnyBranch" is done...but I think you are suggesting that the columns added to repository would be null by default, and would only have values if the user overrides the settings?

Reply:
yes